### PR TITLE
fabric platform driver mgmt refactor

### PR DESCRIPTION
**Pull request by adecaro**
This PR does the following: It refactors the fabric platform driver management to have each driver decide on the dependency to set. Dig must be used just to provide external dependencies.

### CODE REVIEW

File: platform/fabric/sdk/generic/providers.go [hunk 1]
```diff
@@ -28,24 +28,6 @@ import (
...
-	Hasher                  hash.Hasher
-	TracerProvider          trace.TracerProvider
-	Drivers                 []driver5.NamedDriver `group:"db-drivers"`
-	ChannelConfigProvider   driver.ChannelConfigProvider
-	ListenerManagerProvider driver.ListenerManagerProvider
-}) channelProviderResult {
-	return channelProviderResult{Provider: generic.NewProvider(in.KVS, in.Publisher, in.Hasher, in.TracerProvider, in.Drivers, in.ChannelConfigProvider, in.ListenerManagerProvider)}
-}
-
 type ChannelHandlerProviderResult struct {
```
Comment (mbrandenburger) [inline platform/fabric/sdk/generic/providers.go:31]: I remember that Alex suggested to keep all code that depends/uses on dig in a separate package to keep a structure where we see easily when dig is used.

@alexandrosfilios does this strategy still applies?

**Review by mbrandenburger — CHANGES_REQUESTED** [review]

File: docs/fabric/drivers.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# Fabric Platform Driver Architecture
+
+The Fabric platform consists of a [`Fabric API`](https://github.com/hyperledger-labs/tree/main/fabric-smart-client/platform/fabric)
```
[the remark this answers is not in the document]
Comment (adecaro) [reply docs/fabric/drivers.md:3]: okay

**Review by mbrandenburger — APPROVED** [review]
LGTM

**Review by alexandrosfilios — APPROVED** [review]
LGTM

**Pull request merged**